interface/usergroup/usergroup_admin.php

```diff
@@ -301,7 +302,8 @@
             sqlStatement("update users set supervisor_id = ? where id = ? ", array((int)$_POST["supervisor_id"], $_POST["id"]));
         }
         if (isset($_POST["google_signin_email"])) {
-            if (empty($_POST["google_signin_email"])) {
+            // Save email as null if input was empty or invalid email address format (prevent attacks such as stored xss)
+            if (empty($_POST["google_signin_email"]) || !ValidationUtils::isValidEmail($_POST["google_signin_email"]))  {
```

Comment:
We have stayed away from this type of thing (ie. silently fail importing of data if not valid since validation can be tough especially for an application that fully supports internationalization). We generally are very flexible with what can get into the database. For example, the demographics form is filled with places where can insert a stored xss vector, however, we are vigilant on the html/js output side of things to mitigate this with escaping.

Reply:
Google email addresses do not allow Chinese characters for example (screenshot below), this PR adds protections to the UI and this code is just in case an attacker circumvents the UI and sends the request directly (e.g. Postman). Are you ok with that and approving, or would like me to modify some functionality?

![image](https://github.com/openemr/openemr/assets/38800918/7c50d56e-9293-411d-ab82-c9a5ef030b39)

Reply:
Recommend removing the validation check on the server side of things. The protection it provides is not worth the potential bugs that can arise (for example, if the validator doesn't like a valid email address because of a bug or more flexibility in email addresses in the future, then it just simply will not save it without user knowing it; also would silently go away when editing a prior current user for other reasons). Just not worth it. This is why OpenEMR's strategy is to generally escape the output rather than try to validate the input on the server side. Ok to do on the client side since bugs will be obvious and then easily addressed with the central email js function.

Reply:
Sounds good @bradymiller  I have removed the validation check on the server side, it only validates on the client side now. Does this look ok for approval and merging? Thanks!

Reply:
Looks like all automated tests/checks have passed too if you're ok with the changes.

Reply:
Hi @bradymiller and @adunsulag  hope you had a nice weekend. Anything else you need me to change on this PR for approval? Thank you.